File: C# Basics/05.ConditionalStatements/08.DigitAsWord/DigitAsWord.cs
namespace ConditionalStatements
{
    using System;

    /// <summary>
    /// Task 8: Write a program that asks for a digit (0-9), and depending on the input, shows 
    ///         the digit as a word (in English). Print “not a digit” in case of invalid inut. 
    ///         Use a switch statement.
    /// </summary>
    public class DigitAsWord
    {
        public static void Main()
        {
            Console.Title = "Converting digit to name";
            Console.WriteLine("Enter a digit in range [0-9]: ");
            byte number;
            bool isValid = byte.TryParse(Console.ReadLine(), out number);
            if (!isValid)
            {
                number = 10;
            }

            string numberName;
            switch (number)
            {
                case 0:
                    numberName = "zero";
                    break;
                case 1:
                    numberName = "one";
                    break;
                case 2:
                    numberName = "two";
                    break;
                case 3:
                    numberName = "three";
                    break;
                case 4:
                    numberName = "four";
                    break;
                case 5:
                    numberName = "five";
                    break;
                case 6:
                    numberName = "six";
                    break;
                case 7:
                    numberName = "seven";
                    break;
                case 8:
                    numberName = "eight";
                    break;
                case 9:
                    numberName = "nine";
                    break;
                default:
                    numberName = "not a digit";
                    break;
            }

            Console.ForegroundColor = ConsoleColor.Yellow;
            Console.Write("You have entered number: ");
            Console.ForegroundColor = ConsoleColor.Green;
            Console.WriteLine(numberName);
            Console.ForegroundColor = ConsoleColor.White;
            Console.ReadKey();
        }
    }
}

File: JavaScript Basics/02.JavaScript Syntax/11.Digit as Word/digitAsWord.js
// Task 11: Write a JavaScript function convertDigitToWord(value) that prints the name of an integer
//          number n (0<n<10) in English using switch statement. Write JS program digitAsWord.js that
//          prints a few digits on the console. Run the program through Node.js.

(function () {
    "use strict";
    function convertDigitToWord(number) {
        var result;
        if (typeof number !== "number" || !isFinite(number) || number % 1 !== 0 || number < 0 || number > 9) {
            throw 'Invalid number provided!';
        }

        switch (number) {
            case 0:
                result = 'zero';
                break;
            case 1:
                result = 'one';
                break;
            case 2:
                result = 'two';
                break;
            case 3:
                result = 'three';
                break;
            case 4:
                result = 'four';
                break;
            case 5:
                result = 'five';
                break;
            case 6:
                result = 'six';
                break;
            case 7:
                result = 'seven';
                break;
            case 8:
                result = 'eight';
                break;
            case 9:
                result = 'nine';
                break;
        }

        return result;
    }

    var firstNumber = 8,
        secondNumber = 3,
        thirdNumber = 5;

    console.log(convertDigitToWord(firstNumber));
    console.log(convertDigitToWord(secondNumber));
    console.log(convertDigitToWord(thirdNumber));
}());